fabric-object-builders/src/main/java/net/fabricmc/fabric/api/block/BlockSettingsExtensions.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.block;
+
+import net.fabricmc.fabric.mixin.builders.BlockSettingsHooks;
+import net.minecraft.block.Block.Settings;
+import net.minecraft.block.MaterialColor;
+import net.minecraft.item.Item;
+import net.minecraft.sound.BlockSoundGroup;
+import net.minecraft.tag.Tag;
+import net.minecraft.util.Identifier;
+
+public final class BlockSettingsExtensions {
+    private BlockSettingsExtensions() {
+    }
+
+    public static void breakByHand(Settings settings, boolean breakByHand) {
+        FabricBlockSettings.computeExtraData(settings).breakByHand(breakByHand);
+    }
+
+    public static void breakByTool(Settings settings, Tag<Item> tag, int miningLevel) {
```

Comment:
:heart: for the item tag

Reply:
Wait, we do not need mining level... We can use tag reference within another tag

Reply:
This is arranged this way to work cleanly with vanilla's mining level querying APIs. However, I hope to move away from calling breakByTool/breayByHand in BlockSettings eventually.

Reply:
It might be a good idea to have it work by tags, but it could also potentially lead to bloated methods, and there's the worry of people forgetting to put their tools in the proper tag.

Reply:
In short, this is not the PR to be discussing this - this is an existing feature and we can't remove it until 1.15 anyway <o>

Reply:
What does mojang use mining level for?

Reply:
Ore mining, just like us. The main issue is that Mojang's checking for if something is minable with a certain tool level is hardcoded to vanilla blocks, so we have to extend it to add mod support. Tools *do* have an integer mining level in their class.